modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCacheManager.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.credential;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.modules.credential.enums.CredentialType;
+import org.slf4j.MDC;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.Async;
+
+/**
+ * Manage user credential cache.
+ */
+@Slf4j
+@Configuration
+public class CredentialCacheManager {
+
+    private static final Map<String, Map<CredentialType, CredentialDefinition>> CACHES =
+            new ConcurrentHashMap<>();
+
+    /**
+     * Is the cache container empty.
+     *
+     * @return true or false.
+     */
+    public boolean isEmpty() {
+        return CACHES.isEmpty();
+    }
+
+    /**
+     * Put cache.
+     *
+     * @param key   key
```

Comment:
in this entire class.. we must replace "key" with a more concrete value.
One idea would be.. to have a new object which contains - Csp and username. and use that in all method definitions. 
and move the getCacheKey method inside this class.

Reply:
Done.